internal/batch/workflow_test.go

```diff
@@ -13,48 +13,92 @@ import (
 )
 
 func TestBatchActivityStartsProcessingWorkflows(t *testing.T) {
-	// Create a temporary batch directory with two subdirectories.
-	opts := []fs.PathOp{
-		fs.WithDir("transfer1",
-			fs.WithFile("transfer1.txt", ""),
-		),
-		fs.WithDir("transfer2",
-			fs.WithFile("transfer2.txt", ""),
-		),
-	}
-	tmpDir := fs.NewDir(t, "batch", opts...)
-	batchPath := tmpDir.Path()
-	defer tmpDir.Remove()
-
 	// Set up the activity
 	ctrl := gomock.NewController(t)
 	ctx := context.Background()
 	serviceMock := batchfake.NewMockService(ctrl)
 	a := NewBatchActivity(serviceMock)
 
-	// Expectations: the activity starts a processing workflow for each subdirectory.
-	serviceMock.EXPECT().InitProcessingWorkflow(ctx, &collection.ProcessingWorkflowRequest{
-		BatchDir:         batchPath,
-		Key:              "transfer1",
-		IsDir:            true,
-		PipelineNames:    []string{"am"},
-		ProcessingConfig: "automated",
-	})
-	serviceMock.EXPECT().InitProcessingWorkflow(ctx, &collection.ProcessingWorkflowRequest{
-		BatchDir:         batchPath,
-		Key:              "transfer2",
-		IsDir:            true,
-		PipelineNames:    []string{"am"},
-		ProcessingConfig: "automated",
+	t.Run("Starts processing workflows for subdirectories", func(t *testing.T) {
+		// Create a temporary batch directory with two subdirectories.
+		opts := []fs.PathOp{
+			fs.WithDir("transfer1",
+				fs.WithFile("transfer1.txt", ""),
+			),
+			fs.WithDir("transfer2",
+				fs.WithFile("transfer2.txt", ""),
+			),
+		}
+		tmpDir := fs.NewDir(t, "batch", opts...)
+		batchPath := tmpDir.Path()
+		defer tmpDir.Remove()
```

Comment:
This is not needed anymore:

    defer tmpDir.Remove()

In https://pkg.go.dev/gotest.tools/v3/fs#NewDir:

> When used with Go 1.14+ the directory will be automatically removed when the test ends.